datatypes/src/primitives/time_instance.rs

```diff
@@ -0,0 +1,53 @@
+use chrono::{DateTime, TimeZone, Utc, NaiveDateTime};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Deserialize, Serialize, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(C)]
+pub struct TimeInstance(i64);
+
+impl TimeInstance {
+
+    pub fn as_utc_date_time(self) -> DateTime<Utc> {
+        Utc.timestamp_millis(self.0)
+    } 
+
+    pub fn as_naive_date_time(self) -> NaiveDateTime {
+        self.as_utc_date_time().naive_utc()
```

Comment:
I like the into trait but i guess a named method  (which is used by the into call) is more expressive.

Reply:
now i also need the conversion for the as_rfc3339 call. so the trait implementation uses the base implementation.

Reply:
A second thought: Why do we output chrono types here? We could output formatted strings and do calculations within the types.

The problem is that it is not safe to do this conversion since it leads to panics on values out of range.
So if these methods stay, they should return a result by unwinding the panic.

Reply:
done but this introduces an unwrap/expect in the to_rfc_... method.

Reply:
done but this introduces an unwrap/expect in the to_rfc_... method.